R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,161 @@
+#' Summarize Kicking Stats
+#'
+#' @description Build columns that aggregate kicking stats at the game level.
+#'
+#' @param pbp A Data frame of NFL play-by-play data typically loaded with
+#' [load_pbp()] or [build_nflfastR_pbp()].
+#' @param weekly If `TRUE`, returns week-by-week stats, otherwise, stats for
+#' the entire Data frame.
+#'
+#' @examples
+#'
+#' \donttest{
+#' pbp <- nflreadr::load_pbp(2021)
+#' calculate_kicking_stats(pbp, weekly = TRUE)
+#' }
+#'
+#' @return a dataframe of kicking stats
+#' @seealso <https://nflreadr.nflverse.com/reference/load_player_stats.html> for the nflreadr function to download this from repo (`stat_type = "kicking"`)
+#' @export
+calculate_kicking_stats <- function(pbp, weekly = FALSE){
+
+  # First, creating a grouping variable object to toggle the weekly argument w/
+  grp_vars <- list("season", "season_type", "week", "team", "player_id")
+  if(!weekly) {
+    grp_vars <- setdiff(grp_vars, "week")
+  }
+  grp_vars <- lapply(grp_vars, as.symbol)
+
+  # Filtering down / creating a base dataset
+  df_fg_or_pat = pbp %>%
```

Comment:
Can you run styler::style_file on this (noticing the stray equal-assign)

Reply:
Yes--great suggestion! I use `=` for assignment by default and occasionally miss them when need to use `<-`